lib/Common/Memory/HeapConstants.h

```diff
@@ -29,7 +34,25 @@ class HeapConstants
 
     static const uint ObjectAllocationShift = 4;        // 16
     static const uint ObjectGranularity = 1 << ObjectAllocationShift;
+#if USE_STAGGERED_OBJECT_ALIGNMENT_BUCKETS
+    /*
+    Multiplier  Alignment   # of Buckets
+    1               16          48
+    2               32           8
+    4               64           8
+    8               128          8
+    16              256          4
+    Total # of Buckets:         76
```

Comment:
The code looks fine.  Have you measure what the memory impact is?  Is there any other perf impact?